dataland-message-queue-utils/src/main/kotlin/org/dataland/datalandmessagequeueutils/utils/MessageQueueUtils.kt

```diff
@@ -33,4 +34,15 @@ class MessageQueueUtils {
             )
         }
     }
+
+    /**
+     * Extracts a json property from a message in json format
+     * @param message the message as a json string
+     * @param key the key of the property of interest
+     * @returns the value corresponding to the key
+     */
+    fun extractValueFromMessagePayload(message: String, key: String): String {
+        val content = JSONObject(message).toMap()
+        return content[key] as String
+    }
```

Comment:
Ist gesichert, dass das Vorgehen mit `toMap()` stabil ist?
Kann das auch mit nesting umgehen?
Und gibt es in der JSON library nicht eine out-of-the-box Methode, einen Wert auszulesen?

Reply:
There is `JSONObject(message).getString(key)!` for extracting string properties or `JSONObject(message).get(key)!.toString()` for general values. I'd go with the latter one.

However I can not find the method in my code anymore.

Reply:
Yes Siegfried has removed it since it is not necessary anymore.
Therefore closing this conversation.